applications/nrf_desktop/src/modules/ble_bond.c

```diff
@@ -485,13 +486,14 @@ static int settings_set(const char *key, size_t len_rd,
 {
 	if (!strcmp(key, PEER_ID_STORAGE_NAME)) {
 		ssize_t len = read_cb(cb_arg, &cur_peer_id, sizeof(cur_peer_id));
+
 		if (len != sizeof(cur_peer_id)) {
 			LOG_ERR("Can't read peer id from storage");
+			module_set_state(MODULE_STATE_ERROR);
 			return len;
 		}
-	}
-
-	if (!strcmp(key, BT_ID_LUT_STORAGE_NAME)) {
+		__ASSERT_NO_MSG(cur_peer_id < DONGLE_PEER_ID);
```

Comment:
```
if (cur_peer_id >= DONGLE_PEER_ID) {
   ASSERT(false);
   cur_peer_id = 0;
}
```
similarly as in at line 234

Reply:
In line 234 it would not work, `cur_peer_id` could reach value of `DONGLE_PEER_ID` during normal operations.
I would prefer to leave it as it is.